docs/metasploit-framework.wiki/Metasploit-Guide-LDAP.md

```diff
@@ -12,8 +12,114 @@ LDAP on Windows environments are found on:
 
 ### Lab Environment
 
-TODO
+LDAP support is enabled by default on a Windows environment when you install Active Directory.
+For LDAPS support to be enabled on port 636, you will have to configure [[AD CS (Active Directory Certificate Services)|ad-certificates/overview.md]]
+
+### Authentication
+
+The LDAP module supports the following forms of authentication with the `LDAPAuth` option:
+
+- auto
+- ntlm
+- kerberos - Example below
+- plaintext
+- none
 
 ### LDAP Enumeration
 
-TODO
+The `auxiliary/gather/ldap_query.rb` module can be used for querying LDAP:
+
+```
+use auxiliary/gather/ldap_query
+run rhost=192.168.123.13 username=Administrator@domain.local password=p4$$w0rd action=ENUM_ACCOUNTS
+```
+
+Example output:
+
+```
+msf6 auxiliary(gather/ldap_query) > run rhost=192.168.123.13 username=Administrator@domain.local password=p4$$w0rd action=ENUM_ACCOUNTS
+[*] Running module against 192.168.123.13
+
+[*] Discovering base DN automatically
+[+] 192.168.123.13:389 Discovered base DN: DC=domain,DC=local
+CN=Administrator CN=Users DC=domain DC=local
+==========================================
+
+ Name                Attributes
+ ----                ----------
+ badpwdcount         0
+ description         Built-in account for administering the computer/domain
+ lastlogoff          1601-01-01 00:00:00 UTC
+ lastlogon           2023-01-23 11:02:49 UTC
+ logoncount          159
+ memberof            CN=Group Policy Creator Owners,CN=Users,DC=domain,DC=local || CN=Domain Admins,CN=Users,DC=domain,DC=local |
+                     | CN=Enterprise Admins,CN=Users,DC=domain,DC=local || CN=Schema Admins,CN=Users,DC=domain,DC=local || CN=Adm
+                     inistrators,CN=Builtin,DC=domain,DC=local
+ name                Administrator
+ pwdlastset          133189448681297271
+ samaccountname      Administrator
+ useraccountcontrol  512
+
+ ... etc ...
+```
+
+This module has a selection of inbuilt queries which can be configured via the `action` setting to make enumeration easier:
```

Comment:
Thoughts on adding in an example of how to set an action when not using the inline arguments?

Reply:
I'll skip on this now for now, to have consistency of teaching one method to the user